sample_apps/rust/examples/utils/query_common.rs

```diff
@@ -0,0 +1,261 @@
+use aws_sdk_timestreamquery as timestream_query;
+use aws_sdk_timestreamquery::types;
+use aws_types::region::Region;
+use clap::Parser;
+use std::error::Error;
+use std::fs;
+use std::io::Write;
+
+static DEFAULT_DATABASE_NAME: &str = "devops_multi_sample_application";
+static DEFAULT_OUTPUT_FILE: &str = "query_results.log";
+static DEFAULT_REGION: &str = "us-east-1";
+static DEFAULT_TABLE_NAME: &str = "host_metrics_sample_application";
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+pub struct Args {
+    // The Timestream for LiveAnalytics database name to use for all queries
+    #[arg(short, long, default_value = DEFAULT_DATABASE_NAME)]
+    pub database_name: String,
+
+    // The full name of the output file to write query results to.
+    // For example, query_results.log
+    #[arg(short, long, default_value = DEFAULT_OUTPUT_FILE)]
+    pub output_file: String,
+
+    // The name of the AWS region to use for queries
+    #[arg(short, long, default_value = DEFAULT_REGION)]
+    pub region: String,
+
+    // The Timestream for LiveAnalytics table name to use for all queries
+    #[arg(short, long, default_value = DEFAULT_TABLE_NAME)]
+    pub table_name: String,
+}
+
+pub async fn get_connection(
+    region: &String,
+) -> Result<timestream_query::Client, timestream_query::Error> {
+    let config = aws_config::defaults(aws_config::BehaviorVersion::latest())
+        .region(Region::new(region.to_owned()))
+        .load()
+        .await;
+    let (client, reload) = timestream_query::Client::new(&config)
+        .with_endpoint_discovery_enabled()
+        .await
+        .expect("Failure to get the query client connection with Timestream");
+    tokio::task::spawn(reload.reload_task());
+    Ok(client)
+}
+
+pub fn write(mut file: &fs::File, s: String) -> Result<(), Box<dyn Error>> {
+    let s_formatted = format!("{}\n", s);
+    if file.write(s_formatted.as_bytes()).is_err() {
+        return Err(String::from("Failed to write to file").into());
+    }
+    if file.flush().is_err() {
+        return Err(String::from("Failed to flush file").into());
+    }
+    Ok(())
+}
+
+#[allow(dead_code)]
```

Comment:
Should only be for functions that are never called. This aligns with the upstream examples as well so we added the dead code to avoid warnings.

Reply:
The functions are derived from [the goV2 functions](https://github.com/awslabs/amazon-timestream-tools/blob/mainline/sample_apps/goV2/utils/query-common.go), which also doesn't use these functions in any of its sample applications. I believe they are included to show users how results can be parsed and displayed. We don't use any of the functions because it would print too many rows and be cluttered.